insecure/net/network_ingress_test.go

```diff
@@ -0,0 +1,88 @@
+package net
+
+import (
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/golang/protobuf/ptypes/empty"
+	"github.com/stretchr/testify/require"
+
+	"github.com/onflow/flow-go/insecure"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/model/libp2p/message"
+	"github.com/onflow/flow-go/network/channels"
+	"github.com/onflow/flow-go/network/mocknetwork"
+	"github.com/onflow/flow-go/utils/unittest"
+)
+
+// TestHandleIncomingEvent_AttackerRegistered checks that a corrupt network sends ingress messages to a registered attacker.
+// The attacker is mocked out in this test.
+func TestHandleIncomingEvent_AttackerRegistered(t *testing.T) {
+	runCorruptNetworkTest(t, unittest.Logger(),
+		func(
+			corruptedId flow.Identity, // identity of ccf
+			corruptNetwork *Network,
+			adapter *mocknetwork.Adapter, // mock adapter that ccf uses to communicate with authorized flow nodes.
+			stream insecure.CorruptNetwork_ProcessAttackerMessageClient, // gRPC interface that orchestrator network uses to send messages to this ccf.
+		) {
+			codec := unittest.NetworkCodec()
+			attacker := newMockAttacker()
+
+			attackerRegistered := sync.WaitGroup{}
+			attackerRegistered.Add(1)
+			go func() {
+				attackerRegistered.Done()
```

Comment:
```suggestion
				defer attackerRegistered.Done()
```

Reply:
I tried this but it doesn't work because `corruptNetwork.ConnectAttacker()` is a blocking call that waits for a shutdown signal, so the test times out if I use `defer`. The `attackerRegistered.Done()` has to be called before the blocking call.